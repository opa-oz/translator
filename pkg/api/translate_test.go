package api_test

import (
	"bytes"
	"github.com/gkampitakis/go-snaps/snaps"
	"github.com/stretchr/testify/assert"
	"net/http"
	"net/http/httptest"
	"testing"
	"translator/pkg/api"
	"translator/pkg/config"
	"translator/pkg/middlewares"

	"github.com/gin-gonic/gin"
)

func TestGetRoute(t *testing.T) {
	r := gin.Default()
	cfg, _ := config.GetConfig()
	r.Use(middlewares.CfgMiddleware(cfg))

	r.POST("/translate/:fromLang/:toLang", api.Translate)

	t.Run("should translate text", func(t *testing.T) {
		w := httptest.NewRecorder()
		req, _ := http.NewRequest(http.MethodPost, "/translate/ru/en", bytes.NewBuffer([]byte("Привет мир!")))
		r.ServeHTTP(w, req)

		snaps.MatchSnapshot(t, w.Body.String())
		assert.Equal(t, http.StatusOK, w.Code)
	})
}
