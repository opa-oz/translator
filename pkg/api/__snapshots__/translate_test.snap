
[TestGetRoute/should_translate_text - 1]
{"text":"Hello world!"}
---
